src/ctf_evals_core/_util/utils.py

```
@@ -0,0 +1,28 @@
+from typing import Any
+
+
+def get_from_metadata[T](
+    metadata: dict[str, Any] | None, key: str, default: T | None = None
+) -> T | None:
+    """
+    Get a value from metadata, with a default value if it doesn't exist.
+
+    For a ctf task it will first look in the sample_metadata, then the
+    challenge_metadata, then the metadata.
+
+    Args:
+        metadata: The metadata dictionary.
+        key: The key to get.
+        default: The default value to return if the key doesn't exist.
+    """
+    if metadata is None:
+        result: T | None = default
```

Should we raise an exception here? We should fail early when there is a programming error and this condition seems like a programming error.

---

Yeah good point this should never happen